### PR TITLE
chore: update nursery crates to use path deps on `tracing`

### PR DESCRIPTION
**Pull request by hawkw**
This branch updates the "nursery" crates to depend on `tracing` and
`tracing-core` as path dependencies, rather than depending on the
crates.io releases of `tokio-trace` and `tokio-trace-core`. When
`tracing` and `tracing-core` are released, we will change these from
path dependencies to crates.io dependencies.

This branch also updates those crates to track API changes in `tracing`
and `tracing-core`.

Signed-off-by: Eliza Weisman <eliza@buoyant.io>

### CODE REVIEW

File: tracing-env-logger/examples/hyper-echo.rs [hunk 1]
```diff
@@ -124,15 +122,16 @@ fn main() {
...
+        let _enter = server_span.enter();
         let server = tokio::net::TcpListener::bind(&addr)
             .expect("bind")
             .incoming()
             .fold(Http::new(), move |http, sock| {
                 let span = span!(
                     Level::TRACE,
                     "connection",
-                    remote = &field::debug(&sock.peer_addr().unwrap())
+                    remote = %sock.peer_addr().unwrap()
```
Comment (jonhoo) [inline tracing-env-logger/examples/hyper-echo.rs]: Shouldn't this be `?` if it is to stay the same?

File: tracing-tower-http/examples/tower-h2-server.rs [hunk 2]
```diff
@@ -125,54 +121,49 @@ fn main() {
...
-                            .map_err(|e| error!("error {:?}", e))
-                            .and_then(|_| {
-                                debug!("response finished");
-                                future::ok(())
-                            })
-                            .instrument(conn_span2);
-                        reactor.spawn(Box::new(serve));
-
-                        Ok((h2, reactor))
+        let h2 = Server::new(new_svc, Default::default(), reactor.clone());
```
Comment (jonhoo) [inline tracing-tower-http/examples/tower-h2-server.rs]: This call is now no longer inside `serve_span`. Is that intentional?

File: tracing-macros/src/lib.rs [hunk 3]
```diff
@@ -32,11 +32,11 @@ macro_rules! dbg {
             Event, Id, Subscriber,
         };
         let callsite = callsite! {
-            name: concat!("event:trace_dbg(", stringify!($ex), ")"),
+            name: stringify!($ex),
```
Comment (jonhoo) [inline tracing-macros/src/lib.rs]: This seems like an unrelated change?

File: tracing-tower/src/lib.rs [hunk 4]
```diff
@@ -31,19 +31,15 @@ where
...
-        let inner = &mut self.inner;
-        span.enter(|| {
-            // TODO: custom `Value` impls for `http` types would be nice...
-            let span = span!(Level::TRACE, "request", request = &field::debug(&req));
-            let span2 = span.clone();
-            span.enter(move || inner.call(req).instrument(span2))
-        })
+        // TODO: custom `Value` impls for `http` types would be nice...
+        let span =
+            span!(Level::TRACE, parent: &self.span, "request", request = &field::debug(&req));
```
Comment (jonhoo) [inline tracing-tower/src/lib.rs]:
```suggestion
            span!(Level::TRACE, parent: &self.span, "request", request = ?req);
```

Comment (jonhoo) [reply tracing-tower/src/lib.rs]: Also, is the `&` necessary for `parent` here?

File: tracing-env-logger/examples/hyper-echo.rs (hunk 1 above)
Comment (hawkw) [reply tracing-env-logger/examples/hyper-echo.rs]: whoops, will fix that
